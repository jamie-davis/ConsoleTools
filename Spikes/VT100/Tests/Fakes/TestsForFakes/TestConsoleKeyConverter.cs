using System;
using System.Linq;
using System.Reflection.Metadata;
using FluentAssertions.Equivalency;
using Microsoft.VisualStudio.TestPlatform.Utilities;
using TestConsoleLib;
using TestConsoleLib.Testing;
using Xunit;

namespace VT100.Tests.Fakes.TestsForFakes
{
    /// <summary>
    /// Check that the <see cref="ConsoleKeyConverter"/> produces acceptable <see cref="ConsoleKey"/> values.
    /// </summary>
    public class TestConsoleKeyConverter
    {
        [Fact]
        public void CharsAreConvertedToEnumValues()
        {
            //Arrange
            var ranges = new (char Base, char Limit)[]
            {
                ('A', 'Z'),
                ('a', 'z'),
                ('0', '9'),
            };
            var input = ranges.SelectMany(range => Enumerable.Range(range.Base, range.Limit - range.Base))
                .Select(c => (char)c)
                .Concat("\t\r\n[](),.!\"Â£$%^&*+-_=@~#?/<>|\\':;")
                .ToList();

            //Act
            var result = input.Select(i => new { Input = i, Output = ConsoleKeyConverter.FromChar(i) }).ToList();

            //Assert
            var output = new Output();
            output.FormatTable(result);
            output.Report.Verify();
        }
    }
}