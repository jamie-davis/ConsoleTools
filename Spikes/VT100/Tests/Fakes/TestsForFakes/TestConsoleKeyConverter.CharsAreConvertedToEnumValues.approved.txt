Input Output                                                                                   
----- ------------                                                                             
A     A                                                                                        
B     B                                                                                        
C     C                                                                                        
D     D                                                                                        
E     E                                                                                        
F     F                                                                                        
G     G                                                                                        
H     H                                                                                        
I     I                                                                                        
J     J                                                                                        
K     K                                                                                        
L     L                                                                                        
M     M                                                                                        
N     N                                                                                        
O     O                                                                                        
P     P                                                                                        
Q     Q                                                                                        
R     R                                                                                        
S     S                                                                                        
T     T                                                                                        
U     U                                                                                        
V     V                                                                                        
W     W                                                                                        
X     X                                                                                        
Y     Y                                                                                        
a     A                                                                                        
b     B                                                                                        
c     C                                                                                        
d     D                                                                                        
e     E                                                                                        
f     F                                                                                        
g     G                                                                                        
h     H                                                                                        
i     I                                                                                        
j     J                                                                                        
k     K                                                                                        
l     L                                                                                        
m     M                                                                                        
n     N                                                                                        
o     O                                                                                        
p     P                                                                                        
q     Q                                                                                        
r     R                                                                                        
s     S                                                                                        
t     T                                                                                        
u     U                                                                                        
v     V                                                                                        
w     W                                                                                        
x     X                                                                                        
y     Y                                                                                        
0     D0                                                                                       
1     D1                                                                                       
2     D2                                                                                       
3     D3                                                                                       
4     D4                                                                                       
5     D5                                                                                       
6     D6                                                                                       
7     D7                                                                                       
8     D8                                                                                       
      Tab                                                                                      
      Enter                                                                                    
                                                                                               
[     Oem1                                                                                     
]     Oem6                                                                                     
(                                                                                              
)                                                                                              
,     OemComma                                                                                 
.     OemPeriod                                                                                
!                                                                                              
"                                                                                              
Â£                                                                                              
$                                                                                              
%                                                                                              
^                                                                                              
&                                                                                              
*                                                                                              
+     OemMinus                                                                                 
-     OemPlus                                                                                  
_                                                                                              
=     OemPlus                                                                                  
@                                                                                              
~     Oem7                                                                                     
#     End                                                                                      
?     Oem2                                                                                     
/     Oem2                                                                                     
<     OemComma                                                                                 
>     OemPeriod                                                                                
|     Oem5                                                                                     
\     Oem5                                                                                     
'     RightArrow                                                                               
:     Oem1                                                                                     
;     Oem1                                                                                     
                                                                                               